src/hmi2mid.c

```
@@ -0,0 +1,574 @@
+
+/* Sets current to the new event and updates list */
+static void CreateNewEvent(struct hmi_ctx *ctx, int32_t time) {
+    if (!ctx->list) {
+        ctx->list = ctx->current = (midi_event *) calloc(1, sizeof(midi_event));
+        ctx->current->time = (time < 0)? 0 : time;
+        return;
+    }
+
+    if (time < 0) {
+        midi_event *event = (midi_event *) calloc(1, sizeof(midi_event));
+        event->next = ctx->list;
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)